.github/workflows/kind.yaml

```diff
@@ -59,11 +59,78 @@ jobs:
           version: ${{ env.KIND_VERSION }}
           config: ${{ env.KIND_CONFIG }}
 
+      - name: Set NODES_WITHOUT_CILIUM
+        run: |
+          # To add more elements, keep it comma-separated.
+          echo "NODES_WITHOUT_CILIUM=chart-testing-worker2" >> $GITHUB_ENV
+
+      - name: Label nodes
+        if: ${{ matrix.mode == 'helm' }}
+        run: |
+          IFS=',' read -ra nodes <<< "$NODES_WITHOUT_CILIUM"
+          for node in "${nodes[@]}"; do
+            kubectl label nodes "${node}" cilium.io/no-schedule=true
+          done
+
+      # This is needed for the tests that run with nodes without Cilium.
+      - name: Install static routes
+        run: |
+          EXTERNAL_FROM_CIDRS=($(kubectl get nodes -o jsonpath='{range .items[*]}{.spec.podCIDR}{"\n"}{end}'))
+          EXTERNAL_NODE_IPS=() # Nodes IPs are collected to be passed to the Cilium CLI later on.
+
+          # Loop over each pod CIDR from all nodes.
+          for i in "${!EXTERNAL_FROM_CIDRS[@]}"; do
+            EXTERNAL_FROM_CIDR=${EXTERNAL_FROM_CIDRS[i]}
+
+            if [[ $EXTERNAL_FROM_CIDR =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+/[0-9]+$ ]]; then
+              IP_FAMILY="v4"
+            elif [[ $EXTERNAL_FROM_CIDR =~ ^[0-9a-fA-F:]+/[0-9]+$ ]]; then
```

Comment:
Not perfect checks for IP address formats but good enough especially in a workflow where I presume we're controlling the inputs.

Reply:
Yeah, I'm aware they aren't perfect and went for just good enough to be usable. I wanted to avoid complicating the bash code further with super long regexes, but definitely any item for followup if we can also find a way to make it more readable, i.e. using a variable instead and hide the definition of it.